### PR TITLE
Add some type-hints to Model.py

### PR DESCRIPTION
**Pull request by JarriqTheTechie**

### CODE REVIEW

**Comment by JarriqTheTechie** [comment]
When adding multiple return types I will use Union instead of pipe operator.

File: src/masoniteorm/models/Model.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ class Model(TimeStampsMixin, ObservesEvents, metaclass=ModelMeta):
 
     __cast_map__ = {}
 
-    __internal_cast_map__ = {
+    __internal_cast_map__: list[str, Any] = {
```
Comment (josephmancuso) [inline src/masoniteorm/models/Model.py]: is this a list or is it a dict?

**Review by josephmancuso — CHANGES_REQUESTED** [review]

**Comment by josephmancuso** [comment]
also tests are failing

**Comment by JarriqTheTechie** [comment]
> also tests are failing

Tests are failing due to issues with backwards compatibility with older versions of python. That's why I'll use Union types instead of | as a type separator.

**Comment by josephmancuso** [comment]
please fix tests failing. i'd like to merge this in

**Comment by josephmancuso** [comment]
closing bc of tests failing

**Pull request closed without merging**